internal/event/machineevent/machineevent.go

```diff
@@ -0,0 +1,154 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package machineevent
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/gogo/protobuf/proto"
+	irievent "github.com/ironcore-dev/ironcore/iri/apis/event/v1alpha1"
+	irimeta "github.com/ironcore-dev/ironcore/iri/apis/meta/v1alpha1"
+	"github.com/ironcore-dev/libvirt-provider/api"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// Interface for recording events
+type EventRecorder interface {
+	Eventf(log logr.Logger, apiMetadata api.Metadata, eventType string, reason string, messageFormat string, args ...any)
+}
+
+// Interface for listing events
```

Comment:
Comment should have the following format 'EventStore ...' (with an optional leading article)